src/Parallel/GlobalCache.hpp

```diff
@@ -520,6 +560,76 @@ bool GlobalCache<Metavariables>::mutable_global_cache_proxy_is_set() const {
   return mutable_global_cache_ == nullptr;
 }
 
+// For all these functions, if the main proxy is set (meaning we are
+// charm-aware) then just call the sys:: functions. Otherise, use the values set
+// for the testing framework (or the defaults).
+template <typename Metavariables>
+int GlobalCache<Metavariables>::number_of_procs() const {
+  return main_proxy_.has_value()
+             ? sys::number_of_procs()
+             : static_cast<int>(alg::accumulate(procs_per_node_, 0_st));
+}
+
+template <typename Metavariables>
+int GlobalCache<Metavariables>::number_of_nodes() const {
+  return main_proxy_.has_value() ? sys::number_of_nodes()
+                                 : static_cast<int>(procs_per_node_.size());
+}
+
+template <typename Metavariables>
+int GlobalCache<Metavariables>::procs_on_node(const int node_index) const {
+  return main_proxy_.has_value()
+             ? sys::procs_on_node(node_index)
+             : static_cast<int>(
+                   procs_per_node_[static_cast<size_t>(node_index)]);
+}
+
+template <typename Metavariables>
+int GlobalCache<Metavariables>::first_proc_on_node(const int node_index) const {
+  return main_proxy_.has_value()
+             ? sys::first_proc_on_node(node_index)
+             : static_cast<int>(
+                   std::accumulate(procs_per_node_.begin(),
+                                   procs_per_node_.begin() + node_index, 0_st));
+}
+
+template <typename Metavariables>
+int GlobalCache<Metavariables>::node_of(const int proc_index) const {
+  if (main_proxy_.has_value()) {
+    return sys::node_of(proc_index);
```

Comment:
Is there an easy way to test this line? It looks like it is tested in `Test_AlgorithmGlobalCache`, right? If so, do you just want to `// LCOV_EXCL` this line and add a C++ comment that for some reason gcov isn't picking it up as being covered?